### PR TITLE
STAC-N tunings update.

### PR DESCRIPTION
**Pull request by mrsiano**
this patch was made after long research about best tunings effort
for dell-r730 gear.
by this patch we can use that gear for low-latency benchmark properly.

includes:
- update the yaml prepare nodes for tunings.
- update onload configuration
- update grbu2 conf
- disable services
- cpu isolation

### CODE REVIEW

**Comment by mrsiano** [comment]
@openshift/svt

**Comment by mffiedler** [comment]
@schituku  @vikaschoudhary16 please take a look

**Comment by mrsiano** [comment]
@jeremyeder 
I've applied your comments

File: networking/synthetic/stac/tuned.conf [hunk 1]
```diff
@@ -1,25 +1,14 @@
 # tuned configuration
 #
-
 [main]
-include=network-latency
+include=cpu-partitioning
```
Comment (mrsiano) [inline networking/synthetic/stac/tuned.conf]: cpu-partitioning already includes the network-latency

File: networking/synthetic/stac/tuned.conf [hunk 2]
```diff
@@ -1,25 +1,14 @@
...
 
-[cpu]
```
Comment (mrsiano) [inline networking/synthetic/stac/tuned.conf]: we don't need that, since using cpu-partitioning

File: networking/synthetic/stac/tuned.conf [hunk 3]
```diff
@@ -1,25 +1,14 @@
...
 [bootloader]
-# cmdline = YOUR_ADDITIONAL_KERNEL_PARAMETERS
-#cmdline = audit=0 selinux=0 isolcpus=1-15 nohz_full=1-15
-#cmdline = audit=0 selinux=0 isolcpus=1-15 nosoftlockup mce=ignore_ce idle=poll
-cmdline = audit=0 selinux=0 isolcpus=1-15 nosoftlockup mce=ignore_ce
+cmdline = audit=0 selinux=0 isolcpus=${isolated_cores} nosoftlockup mce=ignore_ce skew_tick=1
 
 [sysctl]
 kernel.nmi_watchdog=0
-vm.stat_interval=10
```
Comment (mrsiano) [inline networking/synthetic/stac/tuned.conf]: same here

File: networking/synthetic/stac/tuned.conf [hunk 4]
```diff
@@ -1,25 +1,14 @@
...
-
-[sysfs]
-/sys/bus/workqueue/devices/writeback/cpumask = 1
-
-#[scheduler]
-#group.netperf=0:f:1:20:netperf.*$
-
```
Comment (mrsiano) [inline networking/synthetic/stac/tuned.conf]: same here

File: networking/synthetic/stac/tuned_variables.conf [hunk 5]
```diff
@@ -1,5 +0,0 @@
-# Examples:
```
Comment (mrsiano) [inline networking/synthetic/stac/tuned_variables.conf]: we use cpu-partitioning-variables.conf now

File: networking/synthetic/scripts/tuning.sh [hunk 6]
```diff
@@ -1,25 +0,0 @@
-#!/bin/sh
```
Comment (mrsiano) [inline networking/synthetic/scripts/tuning.sh]: replaced with script.sh, see it below

File: networking/synthetic/stac-prepare-nodes.yaml [hunk 7]
```diff
@@ -1,30 +1,54 @@
...
   - name: Create stac tuned profile directory if not exists
-    file: path=/etc/tuned/stac state=directory
-    
+    file: path={{ tuned_stac_dir }}  state=directory
+
   - name: Copy tuned stac profile to producer and consumer
-    copy: src=stac/tuned.conf dest=/etc/tuned/stac/tuned.conf
+    copy: src=stac/* dest={{ tuned_stac_dir }}/
+
+  - name: set isolated_cores for tuned
```
Comment (mrsiano) [inline networking/synthetic/stac-prepare-nodes.yaml]: "set" with upper case

File: networking/synthetic/stac-prepare-nodes.yaml [hunk 8]
```diff
@@ -1,30 +1,54 @@
...
+    replace:
+      dest={{ tuned_stac_dir }}/cpu-partitioning-variables.conf
+      regexp='<range>'
+      replace={{ isolated_cores }}
 
   - name: Copy stac profile script to producer and consumer
-    copy: src=scripts/tuning.sh dest=/etc/tuned/stac/tuning.sh owner=root group=root mode=744
+    copy: src=stac/script.sh dest=/home/script.sh owner=root group=root mode=744
+
+  - name: set isolated_cores for preperation script
```
Comment (mrsiano) [inline networking/synthetic/stac-prepare-nodes.yaml]: same here

File: networking/synthetic/stac/script.sh [hunk 9]
```diff
@@ -0,0 +1,157 @@
+#!/bin/sh
+
+numa_cores=<range>
+mtu_size="1800"
+
+start() {
+    echo "200" > /proc/sys/vm/nr_hugepages
```
Comment (jeremyeder) [inline networking/synthetic/stac/script.sh]: Move the hugepages allocation to tuned please.

File: networking/synthetic/stac/script.sh [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+    ethtool -C <benchmark_interface> rx-usecs-irq 0 adaptive-rx off
```
Comment (jeremyeder) [inline networking/synthetic/stac/script.sh]: What replaces <benchmark_interface> ?

File: networking/synthetic/stac/tuned.conf [hunk 11]
```diff
@@ -1,25 +1,14 @@
...
+[variables]
+# User is responsible for updating variables.conf with variable content such as isolated_cores=X-Y
+include=/etc/tuned/stac/cpu-partitioning-variables.conf
 
 [bootloader]
-# cmdline = YOUR_ADDITIONAL_KERNEL_PARAMETERS
-#cmdline = audit=0 selinux=0 isolcpus=1-15 nohz_full=1-15
-#cmdline = audit=0 selinux=0 isolcpus=1-15 nosoftlockup mce=ignore_ce idle=poll
-cmdline = audit=0 selinux=0 isolcpus=1-15 nosoftlockup mce=ignore_ce
+cmdline = audit=0 selinux=0 isolcpus=${isolated_cores} nosoftlockup mce=ignore_ce skew_tick=1
```
Comment (jeremyeder) [inline networking/synthetic/stac/tuned.conf]: what is unique about this bootloader line that we have to carry it?  isnt it the same as the one from cpu-partitioning?

File: networking/synthetic/stac/script.sh (hunk 10 above)
Comment (mrsiano) [reply networking/synthetic/stac/script.sh]: "p5p1" we read it from the stac_config file, it is part of the prepare-nodes.py

File: networking/synthetic/stac/tuned.conf (hunk 11 above)
Comment (mrsiano) [reply networking/synthetic/stac/tuned.conf]: I advised with Joe about it and he suggested to keep it.
would you like to ping him about it ?!

File: networking/synthetic/stac/script.sh (hunk 9 above)
Comment (mrsiano) [reply networking/synthetic/stac/script.sh]: ok, will change that

File: networking/synthetic/stac/tuned.conf (hunk 11 above)
Comment (jeremyeder) [reply networking/synthetic/stac/tuned.conf]: keep it, absolutely. but if it is no different than the one that cpu-partitioning would create, then why are you also carrying it in the stac tuned profile?

Comment (mrsiano) [reply networking/synthetic/stac/tuned.conf]: the diffrence is about the cores. so now we inject the right one

**Comment by mrsiano** [comment]
@jeremyeder @vikaschoudhary16 @schituku @mffiedler 
please review

**Comment by jeremyeder** [comment]
@mrsiano I can't review this because I can't test it.  We need someone to test it (other than you).

**Comment by mrsiano** [comment]
@jeremyeder @vikaschoudhary16 
vikas could you test it please ?

**Comment by mffiedler** [comment]
Please reopen if and when we are ready to go forward with this.

**Pull request closed without merging**